### DISCUSSION TITLE
Is it possible to save and reload a notebook programmatically from python?

### DISCUSSION THREAD

**Discussion by Koncopd** [microsoft/vscode-jupyter#14251] [Questions and Answers]
It is possible to do using the classic `jupyter notebook` <7 via
```
from IPython.display import Javascript, display_javascript
js = Javascript("IPython.notebook.save_notebook()")
display_javascript(js)
```
and
```
js = Javascript("IPython.notebook.load_notebook(IPython.notebook.notebook_path)")
display_javascript(js)
```

On jupyter lab it is possible through [ipylab](https://github.com/jtpio/ipylab).
```
from ipylab import JupyterFrontEnd
app = JupyterFrontEnd()
app.commands.execute("docmanager:save")
app.commands.execute("docmanager:reload")
```
Is anything similar possible for `vs code jupyter`? It would be really great to add interactive support for `vs code jupyter` to out project https://github.com/laminlabs/nbproject

**Comment by DonJayamanne** [comment]
Unfortunately that’s not supported, no
As Jupyter lab also doesnt support this we decided not to support this in vscode

**Comment by Koncopd** [comment]
@DonJayamanne thank you for the answer. Is it possible still to do something like `ipylab`? An addon which interacts with python?
